### PR TITLE
[move] Consolidate source/dep logic 

### PR DESCRIPTION
**Pull request by tnowacki**
## Description 

- We have layered on some notions of dependencies/source targets because of the package system. 
- I have created an enum with 3 variants, but I kind of hate the name. So open to suggestions here

## Test plan 

- Internal change 

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: external-crates/move/crates/move-compiler/src/expansion/ast.rs
Comment (tnowacki) [inline external-crates/move/crates/move-compiler/src/expansion/ast.rs:111]: I kind of hate these names, open to suggestions

Comment (cgswords) [reply external-crates/move/crates/move-compiler/src/expansion/ast.rs:111]: I agree, these names don't communicate what is going on very well.

A few suggestions:

1) 
```
pub enum TargetKind {
  Dependency { from_source bool },
  RootProject // or similar
}
```

2) 
```
pub enum TargetKind {
  SourceDependency,
  BytecodeDependency,
  Source
}
```

I also asked ChatGPT:

```
enum PackageType {
    MainProject,
    SourceDependency,
    BytecodeDependency,
}
```

File: external-crates/move/crates/move-compiler/src/sui_mode/id_leak.rs
Comment (cgswords) [inline external-crates/move/crates/move-compiler/src/sui_mode/id_leak.rs:109]:
```suggestion
            // Skip non-source and dependency modules
```

File: external-crates/move/crates/move-compiler/src/sui_mode/typing.rs
Comment (cgswords) [inline external-crates/move/crates/move-compiler/src/sui_mode/typing.rs:118]:
```suggestion
            // Skip non-source and dependency modules
```

**Review by cgswords — APPROVED** [review]
I agree the names are _very_ confusing as they are, so approving assuming they will be fixed before landing. 🙏

File: external-crates/move/crates/move-compiler/src/expansion/ast.rs
Comment (tnowacki) [reply external-crates/move/crates/move-compiler/src/expansion/ast.rs:111]: My issue with "BytecodeDependency" is it makes it feel like it was from an `.mv` file

**Pull request merged**
